transfers/utils.py

```diff
@@ -40,9 +51,14 @@ def _call_url_json(url, params, method='GET'):
         try:
             return response.json()
         except ValueError:  # JSON could not be decoded
-            LOGGER.warning('Could not parse JSON from response: %s',
-                           response.text)
-            return errors.ERR_PARSE_JSON
+            if not response.ok:
+                LOGGER.warning('Could not parse JSON from response: %s',
+                               response.text)
+                return errors.ERR_PARSE_JSON
+            pass
```

Comment:
We are already checking `if not response.ok` on line 46 so I don't think the block in 55-57 can ever be reached. 

Also, it is confusing (though not introduced by this PR) that we have nested `try/except` blocks. Can we separate out these different error conditions?

I think what you're trying to do here is allow for a non-JSON response. So maybe it would be better to have a `json` (or `assume_json`) kwarg in the function that defaults to `True` but when `False` returns `response.text`. How about something like the following?:

    def _call_url_json(url, params, method=METHOD_GET, headers=None,
                    assume_json=True):
        """Helper to GET a URL where the expected response is 200 with JSON.

        :param str url: URL to call
        :param dict params: Params to pass as HTTP query string or JSON body
        :param str method: HTTP method (e.g., 'GET')
        :param dict headers: HTTP headers
        :param bool assume_json: set to False if the response body should not be decoded as JSON
        :returns: Dict of the returned JSON or an integer error
                code to be looked up
        """
        method = method.upper()
        LOGGER.debug('URL: %s; params: %s; method: %s', url, params, method)
        try:
            if method == METHOD_GET:
                response = requests.request(method, url=url, params=params,
                                            headers=headers)
            else:
                response = requests.request(method, url=url, data=params,
                                            headers=headers)
                                            headers=headers)
            LOGGER.debug('Response: %s', response)
            LOGGER.debug('type(response.text): %s ', type(response.text))
            LOGGER.debug('Response content-type: %s',
                        response.headers['content-type'])
        except (urllib3.exceptions.NewConnectionError,
                requests.exceptions.ConnectionError) as err:
            LOGGER.error("Connection error %s", err)
            return errors.ERR_SERVER_CONN
        if not response.ok:
            LOGGER.warning('%s Request to %s returned %s %s', method, url,
                        response.status_code, response.reason)
            LOGGER.debug('Response: %s', response.text)
            return errors.ERR_INVALID_RESPONSE
        if assume_json:
            try:
                return response.json()
            except ValueError:  # JSON could not be decoded
                LOGGER.warning('Could not parse JSON from response: %s',
                            response.text)
                return errors.ERR_PARSE_JSON
        return response.text

Reply:
Aside from the overhead of re-recording our `vcrpy` fixtures noted above, with slight modification I've added it into `utils.py` thanks for taking the time to write this version.